paasta_tools/cli/utils.py

```diff
@@ -305,7 +305,7 @@ def guess_instance(service, cluster, args):
     else:
         try:
             instances = list_all_instances_for_service(
-                service=service, cluster=cluster, instance_type=None, soa_dir=args.yelpsoa_config_root)
```

Comment:
So to clarify (for my sake as well as anyone else reading this), since I think these line comments and lack of context might be causing some confusion.

Previously, list_all_instances_for_service was not actually using its cluster argument (due to a scoping issue). As a result, the cluster argument in guess_instance was not actually doing anything. As part of some cleanup, I removed the argument. Per @solarkennedy's feedback, I added it back to list_all_instances_for_service, but I guess I forgot to update guess_instance. I have now fixed that.